File: particle_simulation.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

typedef struct{
    float x;
    float y;
} Vector2;

class Particle{
    public:
        void PrintParticle();
        void DefineParticle(Vector2 pos, Vector2 vel);
        void RunParticle(int t);
        Vector2 position;
        Vector2 velocity;
};

void Particle::PrintParticle(){
    cout << " X: " << this->position.x << " Y: " << this->position.y << '\n';
}

void Particle::DefineParticle(Vector2 pos, Vector2 vel){
    this->position.x = pos.x;
    this->position.y = pos.y;
    this->velocity.x = vel.x;
    this->velocity.y = vel.y;
}

void Particle::RunParticle(int t){
    this->position.x+=(this->velocity.x*t);
    this->position.y+=(this->velocity.y*t);
}

int main(){
    int n;
    cout << "Type in number of particles: ";
    cin >> n;
    vector<Particle> particles;
    float pos[2];
    float vel[2];
    int t;
    cout << "Type in time for running this simulation: ";
    cin >> t;
    for (int i = 0; i < n; i++){
        cout << "Type in cords of particle: ";
        cin >> pos[0] >> pos[1];
        cout << "Type in velocity of particle: ";
        cin >> vel[0] >> vel[1];
        Particle particle;
        particle.DefineParticle((Vector2){pos[0], pos[1]}, (Vector2){vel[0], vel[1]});
        particles.push_back(particle);
    }
    for (Particle particle: particles){
        particle.PrintParticle();
        particle.RunParticle(t);
        particle.PrintParticle();
    }
}

File: particle_simulation.rs
use std::io;

pub struct Vector2{
    x: f32,
    y: f32,
}

pub struct Particle{
    pos: Vector2,
    vel: Vector2,
}

pub const G: f32 = 9.81;

impl Particle{
    pub fn print_particle(&self){
        println!("X: {}  Y: {}", self.pos.x, self.pos.y);
    }
    pub fn run_particle(&mut self, t: i32){
        self.pos.x = (self.vel.x*t as f32)+((G*t as f32*t as f32)/2.0);
        if ((self.vel.y*t as f32)-((G*t as f32*t as f32)/2.0) <= 0.0){
            self.pos.y = 0; 
        }
        else{
            self.pos.y = (self.vel.y*t as f32)-((G*t as f32*t as f32)/2.0);
        }
    }
}

fn take_input(prompt: &str) -> i32 {
    let mut input = String::new();
    println!("{}", prompt);
    io::stdin().read_line(&mut input).expect("Error: taking input");
    let n: i32  = input.trim().parse().unwrap();
    return n;
}

fn main(){
    let n = take_input("Type in number of particles: ");
    let mut particles: Vec<Particle> = vec![];
    for i in 0..n{
        particles.push(Particle{pos: Vector2 {x: take_input("Type in X of particle: ") as f32, y: take_input("Type in Y of particle: ") as f32}, vel: Vector2 {x: take_input("Type in X speed of particle: ") as f32, y: take_input("Type in Y speed of particle: ") as f32,}});
    } 
    let t = take_input("Type in time for running this simulation: ");
    for i in &mut particles{
        i.print_particle();
        i.run_particle(t);
        i.print_particle();
    }
}
